core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -80,8 +77,8 @@ static void toJson(Snapshot snapshot, JsonGenerator generator) throws IOExceptio
     } else {
       // embed the manifest list in the JSON, v1 only
       generator.writeArrayFieldStart(MANIFESTS);
-      for (ManifestFile file : snapshot.allManifests()) {
-        generator.writeString(file.path());
+      for (String location : snapshot.v1ManifestLocations()) {
```

Comment:
I think this should still be `allManifests`.

Reply:
the entire problem is basically that we have to provide a `FileIO` instance (which we don't necessarily have) when calling `allManifests(io)`. In order to initialize `allManifests` for the V1 code path of creating a Snapshot, one has to provide a `List<ManifestFile>`, which requires a `FileIO` instance (unless I'm missing an important piece of information). Therefore we need to have a way to retrieve a collection of manifest locations without a dependency to `FileIO`. Does that make sense?

Reply:
As discussed in https://github.com/apache/iceberg/pull/5773#discussion_r978390263 I changed this to use a fake FileIO to retrieve the paths